réaménagement  des classes

GregDeveaux opened the pull request:
essai infructueux pour l'instant

GregDeveaux committed:
réaménagement  des classes

essai infructueux pour l'instant

vlidu commented on Game_theLastBattle/Fighters.swift:
```diff
@@ -25,20 +26,33 @@ protocol Guild {
 // -----------------------------------------------------------------------------------------------------
 
-class Characters {
+class Fighter: FighterInTheGuild {
```
Tu pourrais renommer Fighters en Fighter, au singulier, comme tu definis les caracteristiques d'un seul fighter.

GregDeveaux replied:
Ok merci, c'est noté.
Je viens de faire des modifs sur la façon dont sera gérée le naming des fighters, 
via l'initialisateur  du fighter pour l'intégration, ensuite je vais créer une autre fonction au niveau de la class Player (ou Game) qui vérifiera l'unicité du nom.
Est-ce que pour toi, cette piste peut fonctionner ?
Merci a+
Greg

vlidu commented on Game_theLastBattle/Fighters.swift:
```diff
@@ -8,12 +8,13 @@
 // -----------------------------------------------------------------------------------------------------
 
-protocol Guild {
+protocol FighterInTheGuild {
```
Ca pourrait aussi simplement s'appeler FighterProtocol :D

GregDeveaux committed:
créer le nom unique

Essai de mettre la création du nom dans l'initialiseur du fighter.
la vérification de l'unicité du nom se fera au niveau du player après l'appel d'un fighter.

GregDeveaux committed:
nom unique retravaillé

Fonction fonctionne, petit problème avec la variable name des types

GregDeveaux committed:
MAJ

ajout de static sur team player 1 et 2 correctif de chemin

GregDeveaux committed:
Nom unique finalisé :-D

Ajout d'une fonction de copie pour modifier type de réf en type de valeur.
Qui sera pratique pour la suite.

GregDeveaux committed:
Ajout de phrase explicative
